fe/fe-core/src/main/java/org/apache/doris/transaction/TransactionEntry.java

```diff
@@ -261,22 +278,62 @@ public long abortTransaction()
         }
     }
 
+    private void beforeFinishTransaction() {
+        if (isTransactionBegan) {
+            List<Long> tableIds = transactionState.getTableIdList().stream().distinct().collect(Collectors.toList());
+            transactionState.setTableIdList(tableIds);
+            subTransactionStates.sort((s1, s2) -> {
+                if (s1.getSubTransactionType() == SubTransactionType.INSERT
```

Comment:
Why using this order?

Reply:
1. We provide that, each statement cannot read the modifications made by other statements within the same transaction.
2. For delete command, there are 2 implementations,  one is delete condition, one is insert. 
3. If the delete condition is committed after the insert, the delete will work for the insert, for example:
```
begin;
insert into t select * from t1 where id = 1;
delete from t where id = 1;
commit;
```
4. So we make the delete condition commit before the insert.